plugins/inputs/sqlserver/azuresqlqueries.go

```diff
@@ -1276,24 +1276,31 @@ END');
 /* Obtain the previous times the query was called -- they are an array because we are working with a list of DBs in the Managed Instance -- and store them in a temporary table */
 DROP TABLE IF EXISTS #TelegrafCachedDbCollectionTimes;
 CREATE TABLE #TelegrafCachedDbCollectionTimes (
-	[database_id] INT NOT NULL PRIMARY KEY,
+	[logical_database_guid] UNIQUEIDENTIFIER NOT NULL PRIMARY KEY,
 	[collectionTime] DATETIMEOFFSET
 );
 
 DECLARE @XmlQueryData XML = @QueryData;
 INSERT INTO #TelegrafCachedDbCollectionTimes
 SELECT
-	s.value('./id[1]', 'int') AS [database_id],
-	s.value('./collectionTime[1]', 'datetimeoffset') AS [collectionTime]
+	s.value('./id[1]', 'uniqueidentifier') AS [logical_database_guid],
+	s.value('./ct[1]', 'datetimeoffset') AS [collectionTime]
 FROM @XmlQueryData.nodes('/collectionTimes/ts') t(s);
 
 /* Delete records older than <Max INTERVAL_LENGTH_MINUTES> * 2 */
 DELETE FROM #TelegrafCachedDbCollectionTimes WHERE collectionTime <= DATEADD(dd, -2, SYSDATETIMEOFFSET());
 
 DECLARE @SqlText NVARCHAR(MAX) = N'
 
+DECLARE @currDbLogicalGuid UNIQUEIDENTIFIER;
+SELECT @currDbLogicalGuid = logical_database_guid FROM sys.dm_user_db_resource_governance WHERE [database_name] = DB_NAME()
```

Comment:
Any reason to use `database_name = DB_NAME()` vs. `database_id = DB_ID()`? It is a remote chance, but a database could be renamed between the time this query runs and the time we collect.

Please terminate query with a semicolon for consistency.

Reply:
@dimitri-furman, the main reason of doing so is that for some DBs DB_ID() returns  values that differ from the values in the DMV. You can try to execute the following code:

`use master`
`select database_id FROM sys.dm_user_db_resource_governance WHERE [database_name] = 'master'`
`select DB_ID()`

In my environment I received the following values: 32763, 1.
I faced with this issue on a system DB, but I am not sure that such case is impossible with a user DB.
Could you please clarify why the ids are different? Is this a Sql Server issue?

Reply:
I see. This will only happen for system databases, and only on MI. There are local and replicated versions of system databases on MI with different IDs, and unfortunately metadata is not always consistent. It does not impact us because we are not collecting from system databases, so you can match on `database_id`. For user databases, `database_id` in this view will always match DB_ID() for the current database.